ansible/roles/containerd/files/3033.3.x-flatcar-lts/etc/systemd/system/containerd.service

```diff
@@ -0,0 +1,27 @@
+[Unit]
```

Comment:
This and the rest of the systemd unit files in this PR are from https://www.flatcar.org/docs/latest/container-runtimes/use-a-custom-docker-or-containerd-version/

Reply:
Would you be able to share the diffs between these systemd files and the ones that already come with `3033.3.x`?

Reply:
Sure, `prepare-docker` is a brand new unit and `docker.socket` is practically identical (still no harm in overwriting it just to be sure i guess), here is the diff:

<details>
<summary>docker.service</summary>

```bash
diff -y -t upstream_docker.service ../ansible/roles/containerd/files/3033.3.x-flatcar-lts/etc/systemd/system/docker.service       
[Unit]                                                             [Unit]
Requires=torcx.target                                              Requires=torcx.target
After=torcx.target                                                 After=torcx.target
Description=Docker Application Container Engine                    Description=Docker Application Container Engine
Documentation=http://docs.docker.com                            |  After=containerd.service docker.socket network-online.target pr
After=containerd.service docker.socket network-online.target    <
Wants=network-online.target                                        Wants=network-online.target
Requires=containerd.service docker.socket                       |  Requires=containerd.service docker.socket prepare-docker.servic
                                                                >  StartLimitBurst=3
                                                                >  StartLimitIntervalSec=60s

[Service]                                                          [Service]
EnvironmentFile=/run/metadata/torcx                                EnvironmentFile=/run/metadata/torcx
Environment=TORCX_IMAGEDIR=/docker                                 Environment=TORCX_IMAGEDIR=/docker
Type=notify                                                        Type=notify
EnvironmentFile=-/run/flannel/flannel_docker_opts.env              EnvironmentFile=-/run/flannel/flannel_docker_opts.env
Environment=DOCKER_SELINUX=--selinux-enabled=true                  Environment=DOCKER_SELINUX=--selinux-enabled=true
                                                                <
# the default is not to use systemd for cgroups because the del    # the default is not to use systemd for cgroups because the del
# exists and systemd currently does not support the cgroup feat    # exists and systemd currently does not support the cgroup feat
# for containers run by docker                                     # for containers run by docker
ExecStart=/usr/bin/env PATH=${TORCX_BINDIR}:${PATH} ${TORCX_BIN |  Environment=PATH=/opt/bin:/usr/local/sbin:/usr/local/bin:/usr/s
                                                                >  ExecStart=/opt/bin/dockerd --host=fd:// --containerd=/run/docke
ExecReload=/bin/kill -s HUP $MAINPID                               ExecReload=/bin/kill -s HUP $MAINPID
LimitNOFILE=1048576                                                LimitNOFILE=1048576
# Having non-zero Limit*s causes performance problems due to ac    # Having non-zero Limit*s causes performance problems due to ac
# in the kernel. We recommend using cgroups to do container-loc    # in the kernel. We recommend using cgroups to do container-loc
LimitNPROC=infinity                                                LimitNPROC=infinity
LimitCORE=infinity                                                 LimitCORE=infinity
# Uncomment TasksMax if your systemd version supports it.          # Uncomment TasksMax if your systemd version supports it.
# Only systemd 226 and above support this version.                 # Only systemd 226 and above support this version.
TasksMax=infinity                                                  TasksMax=infinity
TimeoutStartSec=0                                                  TimeoutStartSec=0
# set delegate yes so that systemd does not reset the cgroups o    # set delegate yes so that systemd does not reset the cgroups o
Delegate=yes                                                       Delegate=yes
# kill only the docker process, not all processes in the cgroup    # kill only the docker process, not all processes in the cgroup
KillMode=process                                                   KillMode=process
# restart the docker process if it exits prematurely               # restart the docker process if it exits prematurely
Restart=on-failure                                                 Restart=on-failure
StartLimitBurst=3                                               <
StartLimitInterval=60s                                          <

[Install]                                                          [Install]
WantedBy=multi-user.target                                         WantedBy=multi-user.target
```

I moved `StartLimitInterval` and `StartLimitBurst` to `Unit` instead of `Service` as per https://github.com/systemd/systemd/blob/v241/NEWS#L3443
</details>

<details>
<summary>containerd.service</summary>

```bash
diff -y -t upstream_containerd.service ../ansible/roles/containerd/files/3033.3.x-flatcar-lts/etc/systemd/system/containerd.service 
[Unit]                                                             [Unit]
Requires=torcx.target                                              Requires=torcx.target
After=torcx.target                                                 After=torcx.target
Description=containerd container runtime                           Description=containerd container runtime
Documentation=https://containerd.io                             |  After=network.target prepare-docker.service
After=network.target                                            |  Requires=prepare-docker.service

[Service]                                                          [Service]
EnvironmentFile=/run/metadata/torcx                                EnvironmentFile=/run/metadata/torcx
Environment=TORCX_IMAGEDIR=/docker                                 Environment=TORCX_IMAGEDIR=/docker
Delegate=yes                                                       Delegate=yes
Environment=CONTAINERD_CONFIG=/usr/share/containerd/config.toml |  Environment=PATH=${PATH}:/opt/bin
ExecStartPre=mkdir -p /run/docker/libcontainerd                    ExecStartPre=mkdir -p /run/docker/libcontainerd
ExecStartPre=ln -fs /run/containerd/containerd.sock /run/docker    ExecStartPre=ln -fs /run/containerd/containerd.sock /run/docker
ExecStart=/usr/bin/env PATH=${TORCX_BINDIR}:${PATH} ${TORCX_BIN |  Environment=CONTAINERD_CONFIG=/etc/containerd/config.toml
                                                                >  ExecStart=/opt/bin/containerd --config ${CONTAINERD_CONFIG}
KillMode=process                                                   KillMode=process
Type=notify                                                     <
Restart=always                                                     Restart=always
RestartSec=5                                                    <

# (lack of) limits from the upstream docker service unit           # (lack of) limits from the upstream docker service unit
LimitNOFILE=1048576                                                LimitNOFILE=1048576
LimitNPROC=infinity                                                LimitNPROC=infinity
LimitCORE=infinity                                                 LimitCORE=infinity
TasksMax=infinity                                                  TasksMax=infinity

[Install]                                                          [Install]
WantedBy=multi-user.target                                         WantedBy=multi-user.target
```
</details>

<details>
<summary>docker.socket</summary>

```bash
diff -y -t upstream_docker.socket ../ansible/roles/containerd/files/3033.3.x-flatcar-lts/etc/systemd/system/docker.socket
[Unit]                                                             [Unit]
Requires=torcx.target                                              Requires=torcx.target
After=torcx.target                                                 After=torcx.target
Description=Docker Socket for the API                           <
PartOf=docker.service                                              PartOf=docker.service
                                                                >  Description=Docker Socket for the API

[Socket]                                                           [Socket]
ListenStream=/var/run/docker.sock                                  ListenStream=/var/run/docker.sock
SocketMode=0660                                                    SocketMode=0660
SocketUser=root                                                    SocketUser=root
SocketGroup=docker                                                 SocketGroup=docker

[Install]                                                          [Install]
WantedBy=sockets.target                                            WantedBy=sockets.target
                                                                <

```
</details>